### DISCUSSION TITLE
Show lines in SI format

### DISCUSSION THREAD

**Discussion by rbtcollins** [boyter/scc#417] [Ideas]
I would really like to be able to see the counts in SI format rather than 123456789 which is not that easy to parse just by glancing at it.

**Comment by boyter** [comment]
I have looked into this, and the problem comes from localization. Relevant issues is https://github.com/boyter/scc/issues/335

The other issue being fitting it into the space allocated. Although that's less of an issue for smaller repositories.

**Reply by rbtcollins** [reply]
Thanks! The localisation aspect of the numbers doesn't personally concern me. The space issue isn't an issue with SI - which is what I'd like to see most of all - as they will be strictly less space to print :)

**Reply by boyter** [reply]
Ah of course SI units. Not sure how you would expect that to look. Possible to see a mockup for how you would expect it?

There is SI units on the megabytes calculation (although this is also configurable if you check the help.)

**Reply by rbtcollins** [reply]
```
SAS                          1        68       14         0       54          0
Zsh                          1         5        1         1        3          0
───────────────────────────────────────────────────────────────────────────────
Total                     358K      114M       8M       13M      92M         5M
```

Thats taken from the end of an scc output at work, with just a straight conversion from the output to SI - dimensionless units. Adding Loc / B / and other dimensions might be interesting, but not needed to solve the problem I had interpreting things.

**Reply by boyter** [reply]
Ah so you see it being mostly for the totals. That makes a lot more sense to me now.

**Reply by rbtcollins** [reply]
I think it would be surprising to see e.g.

```
....
Zsh                      100001         5        1         1        3          0
───────────────────────────────────────────────────────────────────────────────
Total                     358M      114M       8M       13M      92M         5M
```

rather than

```
Zsh                        100M         5        1         1        3          0
───────────────────────────────────────────────────────────────────────────────
Total                     358M      114M       8M       13M      92M         5M
```

but ultimately as long as the totals are in SI it would solve my problem
